### PR TITLE
Factory boy

### PR DESCRIPTION
**Pull request by talos**
* Newly functional `populate_db` that creates 100 random users with high quality random data by default
* Fixtures now tightly bound to test classes, easier to create

TODO:
* [x] Create random connections between users as well
* [x] Create random shared messages on the activity page
* [x] Get Travis to pull the proper image

NOTES:
* You'll want to `docker-compose pull` before trying to run this, in order to update your image.

### CODE REVIEW

File: app/tests/factories.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+
+
+class UserSkillFactory(SQLAlchemyModelFactory): # pylint: disable=no-init,too-few-public-methods
+
+    class Meta:  # pylint: disable=old-style-class,no-init,too-few-public-methods
+        model = UserSkill
+        sqlalchemy_session = db.session
+
+    name = FuzzyChoice(QUESTIONS_BY_ID.keys())
+    #level = FuzzyChoice([l['score'] for l in LEVELS.values()])
```
Comment (toolness) [inline app/tests/factories.py]: Huh, is there a reason this is commented out? Was it not working for some reason?

Comment (talos) [inline app/tests/factories.py]: Yep, it was throwing a strange error.  You can try reproducing yourself by swapping for the next line.

**Comment by toolness** [comment]
NIce! The tests run fine, but I'm noticing that there are only 64 tests, as opposed to the 66 that are in the most recent merge of this branch from master (f301bbd0d6e1ec31f7b603062a18449a619e26b1). What tests are missing?

Also, there are 3 warnings being raised when running the tests on my end:

```
/usr/local/lib/python2.7/dist-packages/sqlalchemy/sql/sqltypes.py:185: SAWarning: Unicode type received non-unicode bind param value 'paul@lennon.com'. (this warning may be suppressed after 10 occurrences)
  (util.ellipses_string(value),))
/usr/local/lib/python2.7/dist-packages/sqlalchemy/sql/sqltypes.py:185: SAWarning: Unicode type received non-unicode bind param value 'dubya@shrub.com'. (this warning may be suppressed after 10 occurrences)
  (util.ellipses_string(value),))
/usr/local/lib/python2.7/dist-packages/sqlalchemy/sql/sqltypes.py:185: SAWarning: Unicode type received non-unicode bind param value 'sly@stone-less-knowledgea...'. (this warning may be suppressed after 10 occurrences)
  (util.ellipses_string(value),))
```

Usually those are squelched by just making sure the string literal being passed in is unicode (e.g. `u'boop'`). Do you mind fixing those so our test runner results are clean?

File: app/tests/factories.py (hunk 1 above)
Comment (toolness) [inline app/tests/factories.py]: Whoa, that is an extremely weird error... I guess factory boy is doing weird stuff with metaclasses or something? Definitely the weirdest Python traceback I've seen in a long time. :disappointed: 

Anyways, it seems we can avoid it if we define that list comprehension *outside* of the `UserSkillFactory` as a named constant, e.g. `LEVEL_SCORES`, and use that... Would you mind doing that just so we keep things DRY?

Comment (toolness) [inline app/tests/factories.py]: e.g.:

```diff
diff --git a/app/tests/factories.py b/app/tests/factories.py
index a83393a..bc84a55 100644
--- a/app/tests/factories.py
+++ b/app/tests/factories.py
@@ -4,7 +4,7 @@ NoI test factories
 Factories to create test objects
 '''

-from app import QUESTIONS_BY_ID, ORG_TYPES, LOCALES
+from app import QUESTIONS_BY_ID, ORG_TYPES, LOCALES, LEVELS
 from app.models import (db, User, UserSkill, UserJoinedEvent,
                         UserExpertiseDomain, UserLanguage, Email,
                         ConnectionEvent, SharedMessageEvent)
@@ -74,6 +74,8 @@ class UserLanguageFactory(SQLAlchemyModelFactory):
     locale = FuzzyChoice(LOCALES)


+LEVEL_SCORES = [lvl['score'] for lvl in LEVELS.values()]
+
 class UserSkillFactory(SQLAlchemyModelFactory): # pylint: disable=no-init,too-f

     class Meta:  # pylint: disable=old-style-class,no-init,too-few-public-metho
@@ -81,8 +83,7 @@ class UserSkillFactory(SQLAlchemyModelFactory): # pylint: disa
         sqlalchemy_session = db.session

     name = FuzzyChoice(QUESTIONS_BY_ID.keys())
-    #level = FuzzyChoice([l['score'] for l in LEVELS.values()])
-    level = FuzzyChoice([-1, 1, 2, 5])
+    level = FuzzyChoice(LEVEL_SCORES)


 class UserFactory(SQLAlchemyModelFactory): # pylint: disable=no-init,too-few-pu
```

**Comment by toolness** [comment]
Oh man, `populate_db` is awesome now!! Nice job!

It spews a bunch of those [unicode warnings](https://github.com/GovLab/noi2/pull/119#issuecomment-146701444) but once they're fixed for the tests, hopefully they'll be fixed here too.

**Comment by toolness** [comment]
> NIce! The tests run fine, but I'm noticing that there are only 64 tests, as opposed to the 66 that are in the most recent merge of this branch from master (f301bbd0d6e1ec31f7b603062a18449a619e26b1). What tests are missing?

Oh, [looks like](https://github.com/GovLab/noi2/pull/119/files#diff-97301c841e766129dc601dc5916af50eL309) it's `test_helpful_users_works` and `test_nearest_neighbors_works`... Should we file an issue to re-introduce those tests at some point?

**Comment by talos** [comment]
Ah yes, I'd disabled those because they weren't working and we actually don't need the functionality in the new wireframes... thoughts?

**Comment by toolness** [comment]
Ah, that's cool!

**Comment by toolness** [comment]
Cool, so if you can get rid of the unicode warnings and do the [`UserSkillFactory` thing](https://github.com/GovLab/noi2/pull/119#discussion_r41577206), I think this is ready to merge!

File: app/tests/factories.py (hunk 1 above)
Comment (talos) [inline app/tests/factories.py]: Smart!  Thanks for taking a look.  I just saw the error message, and was like... huh?  I'll move the declaration to outside.

Comment (talos) [inline app/tests/factories.py]: Also, that's some slick Github automated markup.

**Comment by talos** [comment]
Fixed the unicode and `LEVEL` issue -- thanks for the tip on the second one!

**Pull request merged**